canopen/pdo/base.py

```diff
@@ -387,7 +389,7 @@ def save(self) -> None:
             logger.info("Setting SYNC start value to %d", self.sync_start_value)
             self.com_record[6].raw = self.sync_start_value
 
-        if self.map is not None:
+        if self.map:
```

Comment:
"is not None"-Check may be more efficient, but it is wrong! This is the Bug: self.map is never set to None, therefore this check will always be True and every PDO will be written. self.map is initialized with [] and also reset with [].

Reply:
Gotcha! I didn't notice that, thanks for the explanation.  However, when the map is empty, we should not skip saving the PDO mapping parameter record completely.  There will be no way to erase the mappings in the device.  So instead of "fixing" the always-true conditional, we should just get rid of it altogether?

Reply:
You are right. Lets break this down in a new bug and discuss it there.